src/structures/Channel.js

```diff
@@ -108,27 +108,91 @@ class Channel extends Base {
   }
 
   /**
-   * Indicates whether this channel is {@link TextBasedChannels text-based}.
+   * Indicates whether this channel is a {@link TextChannel}.
    * @returns {boolean}
    */
   isText() {
-    return 'messages' in this;
+    return this.type === 'GUILD_TEXT';
   }
 
   /**
-   * Indicates whether this channel is {@link BaseGuildVoiceChannel voice-based}.
+   * Indicates whether this channel is a {@link DMChannel}.
+   * @returns {boolean}
+   */
+  isDM() {
+    return this.type === 'DM';
+  }
+
+  /**
+   * Indicates whether this channel is a {@link VoiceChannel}.
    * @returns {boolean}
    */
   isVoice() {
-    return VoiceBasedChannelTypes.includes(this.type);
+    return this.type === 'GUILD_VOICE';
+  }
+
+  /**
+   * Indicates whether this channel is a {@link PartialGroupDMChannel}
+   * @returns {boolean}
+   */
+  isGroupDM() {
+    return this.type === 'GROUP_DM';
+  }
+
+  /**
+   * Indicates whether this channel is a {@link CategoryChannel}.
+   * @returns {boolean}
+   */
+  isCategory() {
+    return this.type === 'GUILD_CATEGORY';
+  }
+
+  /**
+   * Indicates whether this channel is a {@link NewsChannel}
+   * @returns {boolean}
+   */
+  isNews() {
+    return this.type === 'GUILD_NEWS';
+  }
+
+  /**
+   * Indicates whether this channel is a {@link StoreChannel}
+   * @returns {boolean}
+   */
+  isStore() {
+    return this.type === 'GUILD_STORE';
   }
 
   /**
    * Indicates whether this channel is a {@link ThreadChannel}.
    * @returns {boolean}
    */
   isThread() {
-    return ThreadChannelTypes.includes(this.type);
+    return 'autoArchiveDuration' in this;
```

Comment:
Wouldn't `this.type.endsWith('_THREAD')` or atleast `includes('THREAD')` be more suitable according to their [naming](https://github.com/discord/discord-api-docs/blob/master/docs/resources/Channel.md?plain=1#L54-L56)?

Reply:
What if they add more thread types that don’t end with _THREAD? Unlikely, but still possible and I don’t think a thread would ever not have autoArchiveDuration, however I’m happy to revert if the maintainers think I should, just did that for consistency with the isTextBased method

Reply:
A defined `parentId` can also indicate a channel's category. Perhaps a combination of checks is safer

Reply:
I can always revert to the thread channel types check, but I think it’d be best to use the same check across all methods

Reply:
I agree with @papsavas here, checking `autoArchiveDuration` makes this type guard feel out of place... Either use an array of thread types and check if this.type is present (ideal), or check if the type includes THREAD (less ideal but still better)

Reply:
Alright I’ll change these ones to check for the type and keep the isXBased ones checking for a property in the channel